numpy/lib/type_check.py

```diff
@@ -149,9 +154,10 @@ def imag(val):
 
     Returns
     -------
-    out : ndarray
-        Output array. If `val` is real, the type of `val` is used for the
-        output.  If `val` has complex elements, the returned type is float.
+    out : ndarray or scalar
+        The imaginary component of a complex argument. If `val` is real,
+        the type of `val` is used for the output.  If `val` has complex
+        elements, the returned type is float.
```

Comment:
Can't tell at this point, but I am a little surprised that `complex128` would return `float64` if the architecture is capable of supporting `complex128`.